sound/soc/sof/intel/hda-ctrl.c

```diff
@@ -158,16 +158,18 @@ void hda_dsp_ctrl_misc_clock_gating(struct snd_sof_dev *sdev, bool enable)
  */
 int hda_dsp_ctrl_clock_power_gating(struct snd_sof_dev *sdev, bool enable)
 {
+	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
```

Comment:
also the wording is a bit unclear

suggested edit:
But when resuming from Sx, we unconditionally re-enable DMI L1, without taking the work-around into account, which could lead to xruns when a suspended capture stream is restarted.
"